R/pkg/R/DataFrame.R

```diff
@@ -1138,6 +1138,9 @@ setMethod("collect",
                   if (!is.null(PRIMITIVE_TYPES[[colType]]) && colType != "binary") {
                     vec <- do.call(c, col)
                     stopifnot(class(vec) != "list")
+                    # If vec is an vector with only NAs, the type is logical
```

Comment:
if the DataFrame column is of type string, shouldn't it converts to R as character (which can be all NA), even though the column only has NULL (which maps to NA in R)?

it seems with this change it would become logical in R instead of character.